sematic/resolvers/log_streamer.py

```diff
@@ -90,37 +175,40 @@ def _start_log_streamer_out_of_process(
     ----------
     file_path:
         The path to the local log file
+    read_from_file_descriptor:
+        The file descriptor to read from; likely the "read" end of a pipe
     upload_interval_seconds:
         The interval between uploads.
     uploader:
         An optional custom uploader for the log data
 
     Returns
     -------
-    The process doing the logging
+    The process id of the process doing the logging
     """
     kwargs = dict(
         file_path=file_path,
+        read_from_file_descriptor=read_from_file_descriptor,
         upload_interval_seconds=upload_interval_seconds,
         remote_prefix=remote_prefix,
         uploader=uploader,
     )
-    process = multiprocessing.Process(
-        group=None,
-        target=_stream_logs_to_remote_from_file,
-        kwargs=kwargs,
-        daemon=True,
-    )
-    process.start()
-    return process
+    pid = os.fork()
+    if pid > 0:
+        # in parent process
+        return pid
+    else:
+        # in child process
+        _stream_logs_to_remote_from_file_descriptor(**kwargs)  # type: ignore
+        # can't ever reach here; the above is an infinite loop
+        raise RuntimeError("This code should be unreachable!")
```

Comment:
Unnecessary `else`.
```suggestion
    pid = os.fork()

    if pid > 0:
        # in parent process
        return pid
    
    # in child process
    _stream_logs_to_remote_from_file_descriptor(**kwargs)  # type: ignore
    
    # can't ever reach here; the above is an infinite loop
    raise RuntimeError("This code should be unreachable!")
```

Reply:
Nit: reasons to prepare the `kwargs` prior to the call?

Reply:
> reasons to prepare the kwargs prior to the call?

I think this was what made sense in a prior iteration of the code. Simplifying now!